### ISSUE TITLE
ImagesiftBot

### ISSUE THREAD

**Issue by Maikuolan** [CIDRAM/CIDRAM#537]
Spotted these entries earlier today in CIDRAM's logs at a website that I help manage:

<details>
<summary>Click to expand</summary>

```
ID: 1700969345-908787-9851754210
Script version: CIDRAM v3.4.2
Date/Time: Sun, 26 Nov 2023 03:29:05 +0000
IP address: 64.124.8.81
User agent: Mozilla/5.0 (compatible; ImagesiftBot; +imagesift.com)
Signatures count: 1
Signatures reference: 64.124.8.0/24
Why blocked: Cloud service ("Castle Global Inc", L35562:F0, [US])!
Reconstructed URI: https://[REDACTED].com/product-tag/[REDACTED]/
Infractions: 0
ASN lookup: 0
Country code lookup: XX
Request method: GET
Hostname: -
CAPTCHA state: Enabled.

ID: 1700969345-536181-9479155991
Script version: CIDRAM v3.4.2
Date/Time: Sun, 26 Nov 2023 03:29:05 +0000
IP address: 64.124.8.81
User agent: Mozilla/5.0 (compatible; ImagesiftBot; +imagesift.com)
Signatures count: 1
Signatures reference: 64.124.8.0/24
Why blocked: Cloud service ("Castle Global Inc", L35562:F0, [US])!
Reconstructed URI: https://[REDACTED].com/product-tag/[REDACTED]/
Infractions: 0
ASN lookup: 0
Country code lookup: XX
Request method: GET
Hostname: -
CAPTCHA state: Enabled.

ID: 1700969345-179974-9122944169
Script version: CIDRAM v3.4.2
Date/Time: Sun, 26 Nov 2023 03:29:05 +0000
IP address: 64.124.8.81
User agent: Mozilla/5.0 (compatible; ImagesiftBot; +imagesift.com)
Signatures count: 1
Signatures reference: 64.124.8.0/24
Why blocked: Cloud service ("Castle Global Inc", L35562:F0, [US])!
Reconstructed URI: https://[REDACTED].com/robots.txt
Infractions: 0
ASN lookup: 0
Country code lookup: XX
Request method: GET
Hostname: -
CAPTCHA state: Enabled.

ID: 1700969344-632273-8575243750
Script version: CIDRAM v3.4.2
Date/Time: Sun, 26 Nov 2023 03:29:04 +0000
IP address: 64.124.8.81
User agent: Mozilla/5.0 (compatible; ImagesiftBot; +imagesift.com)
Signatures count: 1
Signatures reference: 64.124.8.0/24
Why blocked: Cloud service ("Castle Global Inc", L35562:F0, [US])!
Reconstructed URI: https://[REDACTED].com/robots.txt
Infractions: 0
ASN lookup: 0
Country code lookup: XX
Request method: GET
Hostname: -
CAPTCHA state: Enabled.

ID: 1700969344-632273-8575243771
Script version: CIDRAM v3.4.2
Date/Time: Sun, 26 Nov 2023 03:29:04 +0000
IP address: 64.124.8.81
User agent: Mozilla/5.0 (compatible; ImagesiftBot; +imagesift.com)
Signatures count: 1
Signatures reference: 64.124.8.0/24
Why blocked: Cloud service ("Castle Global Inc", L35562:F0, [US])!
Reconstructed URI: https://[REDACTED].com/product-tag/[REDACTED]/
Infractions: 0
ASN lookup: 0
Country code lookup: XX
Request method: GET
Hostname: -
CAPTCHA state: Enabled.

ID: 1700913343-244253-7187215510
Script version: CIDRAM v3.4.2
Date/Time: Sat, 25 Nov 2023 11:55:43 +0000
IP address: 64.124.8.30
User agent: Mozilla/5.0 (compatible; ImagesiftBot; +imagesift.com)
Signatures count: 1
Signatures reference: 64.124.8.0/24
Why blocked: Cloud service ("Castle Global Inc", L35562:F0, [US])!
Reconstructed URI: https://[REDACTED].com/contact
Infractions: 0
ASN lookup: 0
Country code lookup: XX
Request method: GET
Hostname: -
CAPTCHA state: Enabled.

ID: 1700913342-663068-6606029561
Script version: CIDRAM v3.4.2
Date/Time: Sat, 25 Nov 2023 11:55:42 +0000
IP address: 64.124.8.30
User agent: Mozilla/5.0 (compatible; ImagesiftBot; +imagesift.com)
Signatures count: 1
Signatures reference: 64.124.8.0/24
Why blocked: Cloud service ("Castle Global Inc", L35562:F0, [US])!
Reconstructed URI: https://[REDACTED].com/shop/[REDACTED]/
Infractions: 0
ASN lookup: 0
Country code lookup: XX
Request method: GET
Hostname: -
CAPTCHA state: Enabled.

ID: 1700913342-240541-6183511808
Script version: CIDRAM v3.4.2
Date/Time: Sat, 25 Nov 2023 11:55:42 +0000
IP address: 64.124.8.30
User agent: Mozilla/5.0 (compatible; ImagesiftBot; +imagesift.com)
Signatures count: 1
Signatures reference: 64.124.8.0/24
Why blocked: Cloud service ("Castle Global Inc", L35562:F0, [US])!
Reconstructed URI: https://[REDACTED].com/[REDACTED]/
Infractions: 0
ASN lookup: 0
Country code lookup: XX
Request method: GET
Hostname: -
CAPTCHA state: Enabled.

ID: 1700913342-070011-6012979023
Script version: CIDRAM v3.4.2
Date/Time: Sat, 25 Nov 2023 11:55:42 +0000
IP address: 64.124.8.30
User agent: Mozilla/5.0 (compatible; ImagesiftBot; +imagesift.com)
Signatures count: 1
Signatures reference: 64.124.8.0/24
Why blocked: Cloud service ("Castle Global Inc", L35562:F0, [US])!
Reconstructed URI: https://[REDACTED].com/robots.txt
Infractions: 0
ASN lookup: 0
Country code lookup: XX
Request method: GET
Hostname: -
CAPTCHA state: Enabled.

ID: 1700900752-880579-6823541328
Script version: CIDRAM v3.4.2
Date/Time: Sat, 25 Nov 2023 08:25:52 +0000
IP address: 64.124.8.66
User agent: Mozilla/5.0 (compatible; ImagesiftBot; +imagesift.com)
Signatures count: 1
Signatures reference: 64.124.8.0/24
Why blocked: Cloud service ("Castle Global Inc", L35562:F0, [US])!
Reconstructed URI: https://[REDACTED].com/shop/[REDACTED]/
Infractions: 0
ASN lookup: 0
Country code lookup: XX
Request method: GET
Hostname: -
CAPTCHA state: Enabled.

ID: 1700900752-704759-6647725775
Script version: CIDRAM v3.4.2
Date/Time: Sat, 25 Nov 2023 08:25:52 +0000
IP address: 64.124.8.66
User agent: Mozilla/5.0 (compatible; ImagesiftBot; +imagesift.com)
Signatures count: 1
Signatures reference: 64.124.8.0/24
Why blocked: Cloud service ("Castle Global Inc", L35562:F0, [US])!
Reconstructed URI: https://[REDACTED].com/robots.txt
Infractions: 0
ASN lookup: 0
Country code lookup: XX
Request method: GET
Hostname: -
CAPTCHA state: Enabled.

ID: 1700900752-346999-6289970835
Script version: CIDRAM v3.4.2
Date/Time: Sat, 25 Nov 2023 08:25:52 +0000
IP address: 64.124.8.66
User agent: Mozilla/5.0 (compatible; ImagesiftBot; +imagesift.com)
Signatures count: 1
Signatures reference: 64.124.8.0/24
Why blocked: Cloud service ("Castle Global Inc", L35562:F0, [US])!
Reconstructed URI: https://[REDACTED].com/[REDACTED]/
Infractions: 0
ASN lookup: 0
Country code lookup: XX
Request method: GET
Hostname: -
CAPTCHA state: Enabled.

ID: 1700900752-346999-6289970795
Script version: CIDRAM v3.4.2
Date/Time: Sat, 25 Nov 2023 08:25:52 +0000
IP address: 64.124.8.66
User agent: Mozilla/5.0 (compatible; ImagesiftBot; +imagesift.com)
Signatures count: 1
Signatures reference: 64.124.8.0/24
Why blocked: Cloud service ("Castle Global Inc", L35562:F0, [US])!
Reconstructed URI: https://[REDACTED].com/robots.txt
Infractions: 0
ASN lookup: 0
Country code lookup: XX
Request method: GET
Hostname: -
CAPTCHA state: Enabled.

ID: 1700840843-656909-7599875533
Script version: CIDRAM v3.4.2
Date/Time: Fri, 24 Nov 2023 15:47:23 +0000
IP address: 64.124.8.36
User agent: Mozilla/5.0 (compatible; ImagesiftBot; +imagesift.com)
Signatures count: 1
Signatures reference: 64.124.8.0/24
Why blocked: Cloud service ("Castle Global Inc", L35562:F0, [US])!
Reconstructed URI: https://[REDACTED].com/wp-json/wp/v2/product/1367
Infractions: 0
ASN lookup: 0
Country code lookup: XX
Request method: GET
Hostname: -
CAPTCHA state: Enabled.

ID: 1700840842-962483-6905445074
Script version: CIDRAM v3.4.2
Date/Time: Fri, 24 Nov 2023 15:47:22 +0000
IP address: 64.124.8.36
User agent: Mozilla/5.0 (compatible; ImagesiftBot; +imagesift.com)
Signatures count: 1
Signatures reference: 64.124.8.0/24
Why blocked: Cloud service ("Castle Global Inc", L35562:F0, [US])!
Reconstructed URI: https://[REDACTED].com/product-tag/[REDACTED]/
Infractions: 0
ASN lookup: 0
Country code lookup: XX
Request method: GET
Hostname: -
CAPTCHA state: Enabled.

ID: 1700840842-789145-6732108494
Script version: CIDRAM v3.4.2
Date/Time: Fri, 24 Nov 2023 15:47:22 +0000
IP address: 64.124.8.36
User agent: Mozilla/5.0 (compatible; ImagesiftBot; +imagesift.com)
Signatures count: 1
Signatures reference: 64.124.8.0/24
Why blocked: Cloud service ("Castle Global Inc", L35562:F0, [US])!
Reconstructed URI: https://[REDACTED].com/robots.txt
Infractions: 0
ASN lookup: 0
Country code lookup: XX
Request method: GET
Hostname: -
CAPTCHA state: Enabled.

ID: 1700840840-911968-4854939250
Script version: CIDRAM v3.4.2
Date/Time: Fri, 24 Nov 2023 15:47:20 +0000
IP address: 64.124.8.36
User agent: Mozilla/5.0 (compatible; ImagesiftBot; +imagesift.com)
Signatures count: 1
Signatures reference: 64.124.8.0/24
Why blocked: Cloud service ("Castle Global Inc", L35562:F0, [US])!
Reconstructed URI: https://[REDACTED].com/robots.txt
Infractions: 0
ASN lookup: 0
Country code lookup: XX
Request method: GET
Hostname: -
CAPTCHA state: Enabled.

ID: 1700840840-914469-4857442169
Script version: CIDRAM v3.4.2
Date/Time: Fri, 24 Nov 2023 15:47:20 +0000
IP address: 64.124.8.36
User agent: Mozilla/5.0 (compatible; ImagesiftBot; +imagesift.com)
Signatures count: 1
Signatures reference: 64.124.8.0/24
Why blocked: Cloud service ("Castle Global Inc", L35562:F0, [US])!
Reconstructed URI: https://[REDACTED].com/wp-json/wp/v2/product_tag/155
Infractions: 0
ASN lookup: 0
Country code lookup: XX
Request method: GET
Hostname: -
CAPTCHA state: Enabled.

ID: 1700781716-392160-0335124563
Script version: CIDRAM v3.4.2
Date/Time: Thu, 23 Nov 2023 23:21:56 +0000
IP address: 64.124.8.82
User agent: Mozilla/5.0 (compatible; ImagesiftBot; +imagesift.com)
Signatures count: 1
Signatures reference: 64.124.8.0/24
Why blocked: Cloud service ("Castle Global Inc", L35562:F0, [US])!
Reconstructed URI: https://[REDACTED].com/shop/[REDACTED]/
Infractions: 0
ASN lookup: 0
Country code lookup: XX
Request method: GET
Hostname: -
CAPTCHA state: Enabled.

ID: 1700781715-891932-9834895293
Script version: CIDRAM v3.4.2
Date/Time: Thu, 23 Nov 2023 23:21:55 +0000
IP address: 64.124.8.82
User agent: Mozilla/5.0 (compatible; ImagesiftBot; +imagesift.com)
Signatures count: 1
Signatures reference: 64.124.8.0/24
Why blocked: Cloud service ("Castle Global Inc", L35562:F0, [US])!
Reconstructed URI: https://[REDACTED].com/wp-json/wp/v2/pages/28
Infractions: 0
ASN lookup: 0
Country code lookup: XX
Request method: GET
Hostname: -
CAPTCHA state: Enabled.

ID: 1700781715-414406-9357380646
Script version: CIDRAM v3.4.2
Date/Time: Thu, 23 Nov 2023 23:21:55 +0000
IP address: 64.124.8.82
User agent: Mozilla/5.0 (compatible; ImagesiftBot; +imagesift.com)
Signatures count: 1
Signatures reference: 64.124.8.0/24
Why blocked: Cloud service ("Castle Global Inc", L35562:F0, [US])!
Reconstructed URI: https://[REDACTED].com/robots.txt
Infractions: 0
ASN lookup: 0
Country code lookup: XX
Request method: GET
Hostname: -
CAPTCHA state: Enabled.

ID: 1700781714-010571-7953541328
Script version: CIDRAM v3.4.2
Date/Time: Thu, 23 Nov 2023 23:21:53 +0000
IP address: 64.124.8.82
User agent: Mozilla/5.0 (compatible; ImagesiftBot; +imagesift.com)
Signatures count: 1
Signatures reference: 64.124.8.0/24
Why blocked: Cloud service ("Castle Global Inc", L35562:F0, [US])!
Reconstructed URI: https://[REDACTED].com/robots.txt
Infractions: 0
ASN lookup: 0
Country code lookup: XX
Request method: GET
Hostname: -
CAPTCHA state: Enabled.

ID: 1700781714-010572-7953541392
Script version: CIDRAM v3.4.2
Date/Time: Thu, 23 Nov 2023 23:21:53 +0000
IP address: 64.124.8.82
User agent: Mozilla/5.0 (compatible; ImagesiftBot; +imagesift.com)
Signatures count: 1
Signatures reference: 64.124.8.0/24
Why blocked: Cloud service ("Castle Global Inc", L35562:F0, [US])!
Reconstructed URI: https://[REDACTED].com/robots.txt
Infractions: 0
ASN lookup: 0
Country code lookup: XX
Request method: GET
Hostname: -
CAPTCHA state: Enabled.

ID: 1700781714-010572-7953541288
Script version: CIDRAM v3.4.2
Date/Time: Thu, 23 Nov 2023 23:21:53 +0000
IP address: 64.124.8.82
User agent: Mozilla/5.0 (compatible; ImagesiftBot; +imagesift.com)
Signatures count: 1
Signatures reference: 64.124.8.0/24
Why blocked: Cloud service ("Castle Global Inc", L35562:F0, [US])!
Reconstructed URI: https://[REDACTED].com/product-tag/[REDACTED]/
Infractions: 0
ASN lookup: 0
Country code lookup: XX
Request method: GET
Hostname: -
CAPTCHA state: Enabled.
```

</blockquote>
</details>

"ImagesiftBot" is the crawler for "[Imagesift](https://imagesift.com/)", which by all *appearances*, is a search engine focused on reverse image searching, similar to the reverse image search functionality provided by Google, Bing, Yandex, et al. Neither CIDRAM nor any of its modules currently have any signatures targeting that crawler *specifically*, but because it's based out of Castle Global's ranges, and Castle Global is included in the default signature files, it is, as a result, currently being blocked by CIDRAM.

I'm in two minds as to whether or not CIDRAM *should* be blocking that crawler. On one hand, it *appears* like a legitimate search engine (for reverse image searching). On the other hand, I have a few potential concerns about doing anything about them being blocked: It's owned and operated by [thehive.ai](https://thehive.ai/), a company focused on artificial intelligence, machine learning, etc. I haven't been able to find any page at their website which documents clearly what limitations may be in place for any data collected (e.g., whether images collected are used solely for their reverse image search functionality, or whether those same images might then also be used for machine learning, to power their AI software, which based on what I've seen from how various other AI/ML-based companies often operate, could potentially lead to copyrighted images being utilised for image generation by the customers/users of the AI software those images were fed into), and there are some sections in their [Terms of Service](https://thehive.ai/terms-of-use) (e.g., under "5.4 Your Likeness"), which I imagine could *potentially* be construed in some cases legally as them taking permission to do whatever they want with such material. As to whether or not it *would*: No idea. I don't have any experience or knowledge about them in general to make that judgement.

The hostname for the IP addresses in question usually resolve to some variation of `x.x.x.x.available.above.net`, but unfortunately don't seem to resolve back to the originating hostname, which means a forward-reverse lookup strategy wouldn't be viable as a means of verifying such requests. I *could* potentially just write a new bypass into the default signature bypasses for any requests coming from those ranges which match their exact user agent, which would work, but lacking a viable means of verification means that it could also potentially open the floodgates to various kinds of unwanted traffic from those ranges, as long as that unwanted traffic was to match their exact user agent, in order to be able to utilise the bypass. Looking at the records at AbuseIPDB for [64.124.8.0/24](https://www.abuseipdb.com/check-block/64.124.8.0/24), for example, we can see there are 22+ thousand results currently for that one, tiny range of 256 IPv4 addresses which their crawler shares, and actively coming in anew (the most recent record at the time of my writing this new issue being only 6 minutes old). With that kind of traffic coming from that range, I'm rather reluctant to provide any allowances for it whatsoever.

Your thoughts? Let it remain as it is: They continue to be blocked, due to the range being blocked. Or do something about it?

**Comment by eurobank** [comment]
yeap, we are seeing a lot of real and fake AI scrapers latetly. I don't think you could do something for all of them, since new ones come in every month.

They can be bypassed, so let's leave the admins to do some work about it.

**Comment by mikeruss1** [comment]
I would like to provide a generalised answer, we should be blocking anything we believe is an AI scraper. If they want to use our material they should ask, or web owners specifically authorise it. Horses and stable doors.

**Comment by eurobank** [comment]
@mikeruss1 While i have months now an ai.txt file, it seems not to be respected. 

Also a LOT of ai bots appear, fake ones 

IP address: 52.230.152.112 »
User agent: Mozilla/5.0 AppleWebKit/537.36 (KHTML, like Gecko; compatible; GPTBot/1.0; +https://openai.com/gptbot) »
Signatures reference: Core.php:L3135, auxiliary.yml:AI Bots »
Why blocked: Fake GPTBot, AI Bots! »

real ones, strange ones. I think the admin should checkk the logs and block what he wants. Impossible to automate this.

Personally i block what i find.

`User agent ≅ /(CCBot|ChatGPT\-User|GPTBot|Google\-Extended|anthropic\-ai|Omgilibot|Omgili|FacebookBot)/i

User agent ≅ /(Claude\-Web|ClaudeBot)/i`

**Comment by mikeruss1** [comment]
useful tip

**Comment by Maikuolan** [comment]
Appreciate the responses. :-)

I'll leave it as it is, so it can continue to be blocked.
